source/Cosmos.System2/Graphics/Bitmap.cs

```diff
@@ -52,107 +52,121 @@ public Bitmap(uint Width, uint Height, byte[] pixelData, ColorDepth colorDepth)
-            //now reading size of bits per pixel (1, 4, 8, 24, 32) - bytes 28 - 30
-            stream.Read(_short, 0, 2);
-            ushort pixelSize = (ushort)BitConverter.ToInt16(_short, 0);
-            //TODO: Be able to handle other pixel sizes
-            if (pixelSize != 32)
-            {
-                throw new NotImplementedException("Can only handle 32bit pictures!");
-            }
-            //now reading compression type - bytes 30 -> 34
-            stream.Read(_int, 0, 4);
-            uint compression = (uint)BitConverter.ToInt32(_int, 0);
-            //TODO: Be able to handle compressed files
-            if (compression != 0)
-            {
-                throw new NotImplementedException("Can only handle uncompressed files!");
-            }
-            //now reading total image data size(including padding) - bytes 34 -> 38
-            stream.Read(_int, 0, 4);
-            uint totalImageSize = (uint)BitConverter.ToInt32(_int, 0);
-            //Somehow this is 0 for my test bmp
-
-            #endregion BMP Header
-
-            //Set the bitmap to have the correct values
-            Width = imageWidth;
-            Height = imageHeight;
-            Depth = (ColorDepth)pixelSize;
-
-            rawData = new int[Width * Height * pixelSize / 8];
-
-            #region Pixel Table
-
-            //Calculate padding
-            int paddingPerRow = 0;
-            int pureImageSize = (int)(imageWidth * imageHeight * pixelSize / 8);
-            if (totalImageSize != 0)
-            {
-                int remainder = (int)totalImageSize - pureImageSize;
-                if (remainder < 0) throw new Exception("Total Image Size is smaller than pure image size");
-                paddingPerRow = remainder / (int)imageHeight;
-            }
-            else
-            {
-                //total image size is 0 if it is not compressed
-                paddingPerRow = 0;
-            }
-            //Read data
-            stream.Position = (int)pixelTableOffset;
-            int position = 0;
-            Byte[] pixelData = new byte[pureImageSize];
-            stream.Read(pixelData, 0, pureImageSize);
-            Byte[] pixel = new byte[pixelSize / 8]; //Pixel size is in byte
-
-            for (int x = 0; x < imageWidth; x++)
-            {
-                for (int y = 0; y < imageHeight; y++)
+                #region BMP Header
+
+                Byte[] _int = new byte[4];
+                Byte[] _short = new byte[2];
+                //Assume that we are using the BMP (Windows) header format
+                //I am using http://www.fastgraph.com/help/bmp_header_format.html
+                //and https://upload.wikimedia.org/wikipedia/commons/c/c4/BMPfileFormat.png?1519566101894as as reference
+
+                //reading magic number to identify if BMP file (BM as string - 42 4D as Hex) - bytes 0 -> 2
+                stream.Read(_short, 0, 2);
+                if ("42-4D" != BitConverter.ToString(_short))
```

Comment:
It is not needed to convert the "magic number" in string to check if it is correct, it is more efficient to convert '_short' to a proper short and simply do "==". As you are at it create a BMP_MAGIC_NUMBER constant.